### PR TITLE
DPC-34: _Since Support

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
**Why**

This is follow on to (DPC-34: Since PR)[https://github.com/CMSgov/dpc-app/pull/634] which had to be reverted by this (PR)[https://github.com/CMSgov/dpc-app/pull/722] because deploys to DEV were failing. This PR restores the DPC-34 feature along with a config change that should allow the code to run in DEV. 

**What Changed**

The error in the DEV environment was caused by the API service not finding the BB key-store. Here's an example error from the logs. 

```
7
2020-03-31T14:46:31.663-04:00
{"level":"ERROR","logger":"gov.cms.dpc.bluebutton.BlueButtonClientModule","thread":"main","message":"Could not find keystore at location: {}/../bbcerts/bb.keystore","timestamp":"2020-03-31T18:46:31.656+0000"}
@ingestionTime
1585680391907
@log
755619740999:dpc-dev-api
@logStream
api/dpc-dev-api/f20b38b0de3148b2a75a1136f58b2e28
@message
{"level":"ERROR","logger":"gov.cms.dpc.bluebutton.BlueButtonClientModule","thread":"main","message":"Could not find keystore at location: {}/../bbcerts/bb.keystore","timestamp":"2020-03-31T18:46:31.656+0000"}
@timestamp
1585680391663
level
ERROR
logger
gov.cms.dpc.bluebutton.BlueButtonClientModule
message
Could not find keystore at location: {}/../bbcerts/bb.keystore
thread
main
timestamp
2020-03-31T18:46:31.656+0000
```

The fix was to incorporate a set of configs into the DEV, TEST, PROD-SBX API service conf files for the BBClient.

**Choices Made**

**Tickets closed**:

**Future Work**

**Checklist**

- [x] All tests are passing via `make ci-app` (app change) and `make ci-web` (website change)
- [ ] Swagger documentation has been updated
- [ ] FHIR documentation has been updated
- [ ] Any required dpc-ops changes have a PR submitted and mentioned in this ticket
- [ ] Any manual migration steps are documented, scripts written (where applicable), and tested
- [ ] Before merging, any required dpc-ops changes have been approved and merged into master of the dpc-ops repo

### CODE REVIEW

File: dpc-api/src/main/resources/dev.application.conf [hunk 1]
```diff
@@ -23,6 +23,16 @@ dpc.api {
...
+    bbclient {
+        serverBaseUrl = ${BFD_URL}
+
+        keyStore {
+            location = ${BB_KEYSTORE_LOCATION}
+            defaultPassword = ${BB_KEYSTORE_PASS}
+        }
+    }
+
     attributionURL = "http://backend.dpc-dev.local:8080/v1/"
```
Comment (RickHawesUSDS) [inline dpc-api/src/main/resources/dev.application.conf:36]: This section matches a similar section found in the dpc-aggregator.

https://github.com/CMSgov/dpc-app/blob/49dc30e8cd6abdb29fc5adeb234faf571f516267/dpc-aggregation/src/main/resources/dev.application.conf#L16-L23

File: src/test/EndToEndRequestTest.postman_collection.json [hunk 2]
```diff
@@ -1219,6 +1219,146 @@
...
+				},
+				{
+					"name": "Try export with _since",
+					"event": [
+						{
+							"listen": "test",
+							"script": {
+								"id": "5497fa12-aefd-44c2-9254-2beb51c52e31",
+								"exec": [
+									"// Status should be 204",
```
Comment (embh) [inline src/test/EndToEndRequestTest.postman_collection.json:1231]:
```suggestion
									"// Status should be 202",
```
Or maybe just remove the comment; it's clear what the test is.

**Comment by embh** [comment]
Looks good to me, just noticed one small typo. Should we deploy the branch to dev to make sure the config change is working?

**Comment by RickHawesUSDS** [comment]
> Looks good to me, just noticed one small typo. Should we deploy the branch to dev to make sure the config change is working?

This has been done and it worked fine.

File: src/test/EndToEndRequestTest.postman_collection.json (hunk 2 above)
Comment (RickHawesUSDS) [reply src/test/EndToEndRequestTest.postman_collection.json]: Good catch. There is a lot of 202 vs 204 confusion in the file. Pushed a fix.

File: dpc-queue/src/main/java/gov/cms/dpc/queue/JobQueueCommon.java [hunk 3]
```diff
@@ -44,8 +49,25 @@ public UUID createJob(UUID orgID, String providerID, List<String> patients, List
...
+                                           UUID orgID,
+                                           String providerID,
+                                           List<String> patients,
+                                           List<ResourceType> resourceTypes,
+                                           OffsetDateTime since,
+                                           OffsetDateTime transactionTime) {
+        return new JobQueueBatch(jobID, orgID, providerID, patients, resourceTypes, since, transactionTime);
+    }
+
+    protected List<JobQueueBatch> createEmptyBatch(UUID jobID,
```
[the remark this answers is not in the document]
Comment (dhgreene) [reply dpc-queue/src/main/java/gov/cms/dpc/queue/JobQueueCommon.java:62]: @RickHawesUSDS is there a good refactoring for this?

**Pull request merged**